const FAILURES = [
    # Unexpected character in array
    "[1,2,3/4,5,6,7]",
    # Unexpected character in object
    "{\"1\":2, \"2\":3 _ \"4\":5}",
    # Invalid escaped character
    "[\"alpha\\Œ±\"]",
    "[\"\\u05AG\"]",
    # Invalid 'simple' and 'unknown value'
    "[tXXe]",
    "[fail]",
    "‚àû",
    # Invalid number
    "[5,2,-]",
    "[5,2,+Œ≤]",
    # Incomplete escape
    "\"\\",
    # Control character
    "\"\0\"",
    # Issue #99
    "[\"üçï\"_\"üçï\"",
    "-0-5"
]

@testset for fail in FAILURES
    @test_throws ErrorException JSON.parse(Typ(fail))
end
